CONTRIBUTING_TO_OPENSHIFT_PLUGINS.md

```diff
@@ -0,0 +1,280 @@
+* https://updates.jenkins-ci.org/download/plugins/openshift-sync
+* https://updates.jenkins-ci.org/download/plugins/openshift-login
+
+Although we as of yet had to pay attention to it, the CI jobs over on CloudBee's Jenkins server for our 4 plugins are:
+
+* https://jenkins.ci.cloudbees.com/job/plugins/job/openshift-pipeline-plugin/
+* https://jenkins.ci.cloudbees.com/job/plugins/job/openshift-client-plugin/
+* https://jenkins.ci.cloudbees.com/job/plugins/job/openshift-sync-plugin/
+* https://jenkins.ci.cloudbees.com/job/plugins/job/openshift-login-plugin/
+
+These kick in when we cut the official version at the Jenkins Update Center for a given plugin.
+
+
+### Set up local repository to cut release:
+
+To cut a new release of any of our 4 plugins, you'll set up a local clone of the jenkinsci repository, like [https://github.com/jenkinsci/openshift-pipeline-plugin](https://github.com/jenkinsci/openshift-pipeline-plugin),
+and then transfer the necessary commits from the analogous openshift repository, like [https://github.com/openshift/jenkins-plugin](https://github.com/openshift/jenkins-plugin).
+
+Two approaches have evolved with the four plugins:
+
+#### Commits between openshift and jenkinsci repositories are in sync
+
+This condition still exists for the openshift-pipeline and openshift-login plugins.  As such, the process and maintenance is a bit simpler, as we can `git rebase`.
+Set up your git remotes so origin is the jenkinsci repository, and upstream is the openshift repository.  Using openshift-pipeline as an example (and substitute the 
+other plugin names if working with those plugins):
+
+1. From the parent directory you've chosen for you local repository, run `git clone git@github.com:jenkinsci/openshift-pipeline-plugin.git`
+1. Change directories into `openshift-pipeline-plugin`, and run `git remote add upstream git://github.com/openshift/jenkins-plugin`
+1. Then pull and rebase the latest changes from [https://github.com/openshift/jenkins-plugin](https://github.com/openshift/jenkins-plugin) with the following:
+
+```
+	$ git checkout master	
+	$ git fetch upstream	
+	$ git fetch upstream --tags	
+	$ git rebase upstream/master	
+	$ git push origin master	
+	$ git push origin --tags
+```
+
+#### Commits between openshift and jenkinsci repositories are no longer in sync
+
+This condition now exists for the openshift-sync and openshift-client plugins.  As such, we have to `git cherry-pick`.  The recipe is same as above, except the 
+`git rebase upstream/master` is replaced with either a `git cherry-pick <commit id>` or `git cherry-pick -m 1 <merge commit id>` based on whether you are pulling 
+in a direct commit or merge commit.
+
+### Why the commits may be out of sync / Why the separate repositories / How to interpret the differences
+
+#### Testing:
+
+* We can employ our CI server and extended test framework for respositories in https://github.com/openshift.  We cannot for https://github.com/jenkinsci.
```

Comment:
"For this reason we maintain the openshift org repository so we can run CI against incoming changes before publishing them to the "official" jenkinsci repository"